pkg/controller/controller_instance.go

```diff
@@ -1545,10 +1545,16 @@ func (c *controller) prepareServiceInstanceLastOperationRequest(instance *v1beta
 		return nil, err
 	}
 
+	// allow for nil ServicePlan which happens if user sets to non-existing plan
+	planID := ""
+	if servicePlan != nil {
+		planID = servicePlan.Spec.ExternalID
```

Comment:
Shouldn't the plan ID come from `instance.Status.InProgressProperties.ClusterServicePlanExternalID` instead?

Reply:
looks like that will work.  The type's comments though say it will be nil during deprovisioning though:
```golang
	// InProgressProperties is the properties state of the ServiceInstance when
	// a Provision or Update is in progress. If the current operation is a
	// Deprovision, this will be nil.
	InProgressProperties *ServiceInstancePropertiesState `json:"inProgressProperties,omitempty"`
```

Reply:
I took a quick look at the deprovision code and that comment might no longer be true. All last operation requests to the broker should contain the service&plan ID that were sent in the provision/deprovision/bind/unbind request that initiated the async operation. Because users can modify the spec afterwards (while the async operation is in progress), the controller should never source the service&plan ID from the spec.  It should always come from `InProgressProperties`. And if there are any cases where we don't store `InProgressProperties`, we should change that.